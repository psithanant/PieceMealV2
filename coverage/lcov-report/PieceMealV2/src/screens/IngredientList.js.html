<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for PieceMealV2/src/screens/IngredientList.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">PieceMealV2/src/screens</a> IngredientList.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">45.45% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/22</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">28.57% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">45.45% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>10/22</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { ListView, View, Text } from 'react-native';
import { Spinner, Button, Icon } from 'native-base';
import { connect } from 'react-redux';
import IngredientItem from '../components/Ingredient/IngredientItem';
import { getIngredients } from '../actions/index';
&nbsp;
const MyButton = <span class="fstat-no" title="function not covered" >(p</span>rops) =&gt; {
<span class="cstat-no" title="statement not covered" >  return (</span>
    &lt;Button
        onPress={<span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >          if (props.loggedIn) {</span>
<span class="cstat-no" title="statement not covered" >            props.navigation.navigate('AddIngredient');</span>
          } else {
<span class="cstat-no" title="statement not covered" >            props.navigation.navigate('logIn');</span>
          }
        }}
        transparent
    &gt;
        &lt;Icon name="add" size={35} style={{ color: '#FFFFFF' }} /&gt;
    &lt;/Button&gt;
  );
};
&nbsp;
const ConnectedMyButton = connect(<span class="fstat-no" title="function not covered" >(s</span>tate) =&gt; {
<span class="cstat-no" title="statement not covered" >  return {</span>
    loggedIn: state.loginReducer.user != null
  };
})(MyButton);
&nbsp;
export class IngredientResultsList extends Component {
  static navigationOptions = <span class="fstat-no" title="function not covered" >({</span> navigation }) =&gt; {
<span class="cstat-no" title="statement not covered" >    return {</span>
      headerTitleStyle: {
   /* this only styles the title/text (font, color etc.)  */
      },
      headerStyle: {
   /* this will style the header, but does NOT change the text */
        backgroundColor: '#68BAA7'
      },
&nbsp;
      title: 'Ingredients',
      headerRight: &lt;ConnectedMyButton navigation={navigation} /&gt;,
      tabBarIcon: <span class="fstat-no" title="function not covered" >({</span> tintColor }) =&gt; (
<span class="cstat-no" title="statement not covered" >       &lt;Icon name='nutrition' style={{ color: '#68BAA7' }}/&gt;</span>
      ),
      mode: 'modal'
    };
  };
&nbsp;
&nbsp;
  componentWillMount() {
    return this.loadDataSource();
    // return this.props.getIngredients();
  }
&nbsp;
  componentDidMount() {
    this.loadDataSource();
    this.props.getIngredients();
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  co</span>mponentDidUpdate() {
<span class="cstat-no" title="statement not covered" >    this.loadDataSource();</span>
  }
&nbsp;
  loadDataSource() {
      const ds = new ListView.DataSource({
          rowHasChanged: <span class="fstat-no" title="function not covered" >(r</span>1, r2) =&gt; <span class="cstat-no" title="statement not covered" >r1 !== r2</span>
      });
&nbsp;
      this.dataSource = ds.cloneWithRows(this.props.list);
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  re</span>nderRow(rowData) {
<span class="cstat-no" title="statement not covered" >    return &lt;IngredientItem</span>
       rowData={rowData}
       onPress={<span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >         this.props.navigation.navigate('ViewIngredient',</span>
           { ingredient: rowData })
       }} /&gt;;
  }
&nbsp;
&nbsp;
    render() {
      this.loadDataSource();
        return (
          &lt;View style={{ flex: 1 }}&gt;
          {this.props.isFetching &amp;&amp; &lt;Spinner color="green" /&gt; }
          {this.props.list.length &gt; 0 &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >&lt;ListView</span>
            removeClippedSubviews={false}
            dataSource={this.dataSource}
            renderRow={this.renderRow.bind(this)}
            enableEmptySections
          /&gt;}
          &lt;/View&gt;
        );
    }
}
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >(s</span>tate) =&gt; {
<span class="cstat-no" title="statement not covered" >    return {</span>
      list: state.ingredients.ingredients,
      isFetching: state.ingredients.isFetching,
      user: state.loginReducer.user
    };
};
&nbsp;
export default connect(mapStateToProps, { getIngredients })(IngredientResultsList);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Mon Jun 19 2017 17:51:46 GMT-0700 (PDT)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
