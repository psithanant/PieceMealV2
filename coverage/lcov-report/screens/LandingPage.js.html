<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for screens/LandingPage.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">screens</a> LandingPage.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">45.45% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">45.45% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/11</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Container, Icon, Input, InputGroup, Button } from 'native-base';
import SearchResults from '../components/SearchResults';
import { createSearchAction } from '../actions/index';
&nbsp;
export class LandingPage extends Component {
 static navigationOptions = <span class="fstat-no" title="function not covered" >({</span> navigation }) =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
     title: 'Piece Meal',
     tabBarIcon: <span class="fstat-no" title="function not covered" >({</span> tintColor }) =&gt; (
<span class="cstat-no" title="statement not covered" >      &lt;Icon name='home' /&gt;</span>
     )
 });
&nbsp;
 componentDidMount() {
   this.props.searchRecipe('');
 }
&nbsp;
 render() {
   return (
    &lt;Container&gt;
       &lt;InputGroup borderType="regular"&gt;
         &lt;Icon name="md-search" /&gt;
         &lt;Input
           placeholder="Search"
           onChangeText={<span class="fstat-no" title="function not covered" >(n</span>ewText) =&gt; <span class="cstat-no" title="statement not covered" >this.props.searchRecipe(newText)}</span>
         /&gt;
         &lt;Button transparent&gt;
             &lt;Icon name='ios-options' /&gt;
         &lt;/Button&gt;
       &lt;/InputGroup&gt;
       &lt;SearchResults navigation={this.props.navigation}
                      recipes={this.props.recipes}
                      ingredients={this.props.ingredients}
                      ingredients={this.props.ingredients}
       /&gt;
     &lt;/Container&gt;
   );
 }
}
&nbsp;
&nbsp;
const mapStateToPropsLandingPage = <span class="fstat-no" title="function not covered" >(s</span>tate) =&gt; {
<span class="cstat-no" title="statement not covered" > return {</span>
   recipes: state.searchRecipe.recipes,
   ingredients: state.searchRecipe.ingredients
&nbsp;
 };
};
&nbsp;
const mapDispatchtoPropsLandingPage = <span class="fstat-no" title="function not covered" >(d</span>ispatch) =&gt; {
<span class="cstat-no" title="statement not covered" > return {</span>
   searchRecipe: <span class="fstat-no" title="function not covered" >(r</span>ecipes) =&gt; {
<span class="cstat-no" title="statement not covered" >     dispatch(createSearchAction(recipes));</span>
   }
 };
};
&nbsp;
const ConnectedLandingPage = connect(
 mapStateToPropsLandingPage, mapDispatchtoPropsLandingPage)(LandingPage);
&nbsp;
export default ConnectedLandingPage;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Fri Jun 16 2017 14:33:43 GMT-0700 (PDT)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
