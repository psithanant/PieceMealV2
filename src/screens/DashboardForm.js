import React, { Component } from 'react';
import {
  Container,
  Icon,
  Tab,
  Tabs
} from 'native-base';
import { connect } from 'react-redux';
import Profile from '../components/Profile';
import Charts from '../components/Charts';
import FavoriteRecipes from '../components/Recipe/FavoriteRecipes';

export class TableExample extends Component {
  static navigationOptions = ({ navigation }) => {
    return {
      headerTitleStyle: {
   /* this only styles the title/text (font, color etc.)  */
      },
      headerStyle: {
   /* this will style the header, but does NOT change the text */
        backgroundColor: '#68BAA7'
      },
      tabBarIcon: ({ tintColor }) => (<Icon style={{ color: '#68BAA7' }} name='paper' />)
    };
  };

  // componentWillUpdate() {
  //   if (!this.props.user) {
  //     return this.props.navigation.navigate('logIn');
  //   }
  // }

  render() {
    console.log('this is props', this.props);
    const { user } = this.props;
    return (
       <Container>
        {/* <Header hasTabs /> */}
        {/* <Tabs renderTabBar={() => <ScrollableTab />}> */}
        {user && <Tabs locked>
          <Tab heading="Profile" textStyle={{ fontSize: 12 }} activeTextStyle={{ fontSize: 12 }}>
            <Profile />
          </Tab>
          {/* <Tab heading="Statistics" textStyle={{ fontSize: 12 }} activeTextStyle={{ fontSize: 12 }}>
            <Charts />
          </Tab> */}
          <Tab heading="Ingredients" textStyle={{ fontSize: 11 }} activeTextStyle={{ fontSize: 10 }}>
            {/* <FavoriteIngredients /> */}
          </Tab>
          <Tab heading="Recipes" textStyle={{ fontSize: 12 }} activeTextStyle={{ fontSize: 12 }}>
            {/* <Tab2 /> */}
            <FavoriteRecipes />
          </Tab>
          {/* <Tab heading="Friends" textStyle={{ fontSize: 12 }} activeTextStyle={{ fontSize: 12 }}>
          </Tab> */}
        </Tabs> }
      </Container>
    );
  }
}

const mapStateToProps = (state) => {
    return {
      user: state.loginReducer.user || state.signupReducer.user
    };
};

export default connect(mapStateToProps)(TableExample);
