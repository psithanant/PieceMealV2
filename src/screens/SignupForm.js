import React, { Component } from 'react';
import { connect } from 'react-redux';
import Toast from 'react-native-simple-toast';
import { Image } from 'react-native';
import {
    Button,
    Container,
    Content,
    Form,
    Icon,
    Item,
    Input,
    Spinner,
    Text,
} from 'native-base';
import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome';
import {
  firstNameChange,
  lastNameChange,
  emailChange,
  passwordChange,
  confirmChange,
  signupUser
} from '../actions';

export class SignupForm extends Component {
  // static navigationOptions = ({ navigation, header }) => ({
  //   title: 'Sign Up',
  //   mode: 'modal'
  // });

    static navigationOptions = ({ navigation }) => {
      return {
        headerTitleStyle: {
     /* this only styles the title/text (font, color etc.)  */
          color: '#373737'
        },
        headerStyle: {
     /* this will style the header, but does NOT change the text */
          backgroundColor: '#68BAA7'
        },
        headerTintColor: {
      /* this will color your back and forward arrows or left and right icons */
        },
        tabBarIcon: ({ tintColor }) => (
         <Icon name='person-add' style={{ color: '#68BAA7' }} />
       ),
       title: 'Sign Up'
      };
    };

    onFirstNameChange(text) {
        this.props.firstNameChange(text);
    }

    onLastNameChange(text) {
        this.props.lastNameChange(text);
    }

    onEmailChange(text) {
        this.props.emailChange(text);
    }

    onPasswordChange(text) {
        this.props.passwordChange(text);
    }

    onConfirmChange(text) {
        this.props.confirmChange(text);
    }

    onButtonPress() {
      const { passwordMatch, firstName, lastName, email, password, confirm } = this.props;
      if (!passwordMatch) {
          Toast.show("Passwords don't match", Toast.SHORT);
      } else {
          this.props.signupUser({ firstName, lastName, email, password, confirm })
          .then(res => {
            if (res.value) {
              Toast.show(`Thanks for signing up ${res.value.first_name}!`, Toast.SHORT);
            }
            return this.props.navigation.navigate('Dashboard');
          });
      }
    }

    render() {
      console.log('this is signup props', this.props);
        return (
            <Container>
              <Image style={styles.containerStyle} source={require('../assets/appBackgound.png')}>
                <Content>
                    <Form>
                        <Item
                          regular
                          style={{ ...styles.itemStyle, marginTop: 50 }}
                        >
                            <Input
                              label='First Name'
                              placeholder='First Name'
                              value={this.props.firstName}
                              onChangeText={this.onFirstNameChange.bind(this)}
                            />
                        </Item>
                        <Item regular style={styles.itemStyle}>
                            <Input
                              label='Last Name'
                              placeholder='Last Name'
                              value={this.props.lastName}
                              onChangeText={this.onLastNameChange.bind(this)}
                            />
                        </Item>
                        <Item regular style={styles.itemStyle}>
                            <Input
                              label='Email'
                              placeholder='Email'
                              value={this.props.email}
                              onChangeText={this.onEmailChange.bind(this)}
                              keyboardType='email-address'
                              autoCapitalize='none'
                              autoCorrect={false}
                            />
                        </Item>
                        <Item regular style={styles.itemStyle}>
                            <Input
                              error={!this.props.passwordMatch}
                              secureTextEntry
                              label='Password'
                              placeholder='Password'
                              value={this.props.password}
                              onChangeText={this.onPasswordChange.bind(this)}
                            />
                        </Item>
                        <Item regular style={styles.itemStyle}>
                            <Input
                              error={!this.props.passwordMatch}
                              secureTextEntry
                              label='Confirm'
                              placeholder='Enter Password Again'
                              value={this.props.confirm}
                              onChangeText={this.onConfirmChange.bind(this)}
                            />
                        </Item>
                          {this.props.loading && <Spinner />}
                        <Button
                          style={styles.buttonStyle}
                          name="email" block padder
                          onPress={() => this.onButtonPress()}
                        >
                          <Icon name='send' />
                            <Text>Sign Up With Email</Text>
                        </Button>
                    </Form>
                </Content>
              </Image>
            </Container>
        );
    }
}

const styles = {
  containerStyle: {
    flex: 1,
    width: undefined,
    height: undefined,
    backgroundColor: 'transparent',
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttonStyle: {
    marginLeft: 88,
    marginRight: 100,
    marginBottom: 25,
    marginTop: 20,
    width: '50%',
    backgroundColor: '#C0B083',
    // alignItems: 'center'
  },
  itemStyle: {
    marginLeft: 25,
    marginRight: 25,
    marginBottom: 10,
    marginTop: 0
  },
};

const mapStateToProps = (state) => {
    return {
      firstName: state.signupReducer.firstName,
      lastName: state.signupReducer.lastName,
      email: state.signupReducer.email,
      password: state.signupReducer.password,
      confirm: state.signupReducer.confirm,
      loading: state.signupReducer.loading,
      passwordMatch: state.signupReducer.passwordMatch
    };
};

export default connect(mapStateToProps, { firstNameChange, lastNameChange, emailChange, passwordChange, confirmChange, signupUser })(SignupForm);
