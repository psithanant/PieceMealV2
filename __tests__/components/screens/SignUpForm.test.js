import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import { SignupForm } from '../../../src/screens/SignupForm';

describe('Sign Up form', () => {
  it('renders SignUpForm component', () => {
  const component = shallow(
    <SignupForm
      firstNameChange={jest.fn}
      lastNameChange={jest.fn}
      emailChange={jest.fn}
      passwordChange={jest.fn}
      confirmChange={jest.fn}
      signupUser={jest.fn}
      firstName={'firstName'}
      lastName={'lastName'}
      email={'email'}
      password={'password'}
      confirm={'confirm'}
      passwordMatch={true}
    />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
