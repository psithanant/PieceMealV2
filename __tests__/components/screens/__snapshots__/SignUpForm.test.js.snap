// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign Up form renders SignUpForm component 1`] = `
<Styled(Container)>
  <Image
    source={1}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "height": undefined,
        "justifyContent": "center",
        "width": undefined,
      }
    }
  >
    <Styled(Content)>
      <Styled(Form)>
        <Styled(Item)
          regular={true}
          style={
            Object {
              "marginBottom": 10,
              "marginLeft": 25,
              "marginRight": 25,
              "marginTop": 50,
            }
          }
        >
          <Styled(Input)
            label="First Name"
            onChangeText={[Function]}
            placeholder="First Name"
            value="firstName"
          />
        </Styled(Item)>
        <Styled(Item)
          regular={true}
          style={
            Object {
              "marginBottom": 10,
              "marginLeft": 25,
              "marginRight": 25,
              "marginTop": 0,
            }
          }
        >
          <Styled(Input)
            label="Last Name"
            onChangeText={[Function]}
            placeholder="Last Name"
            value="lastName"
          />
        </Styled(Item)>
        <Styled(Item)
          regular={true}
          style={
            Object {
              "marginBottom": 10,
              "marginLeft": 25,
              "marginRight": 25,
              "marginTop": 0,
            }
          }
        >
          <Styled(Input)
            autoCapitalize="none"
            autoCorrect={false}
            keyboardType="email-address"
            label="Email"
            onChangeText={[Function]}
            placeholder="Email"
            value="email"
          />
        </Styled(Item)>
        <Styled(Item)
          regular={true}
          style={
            Object {
              "marginBottom": 10,
              "marginLeft": 25,
              "marginRight": 25,
              "marginTop": 0,
            }
          }
        >
          <Styled(Input)
            error={false}
            label="Password"
            onChangeText={[Function]}
            placeholder="Password"
            secureTextEntry={true}
            value="password"
          />
        </Styled(Item)>
        <Styled(Item)
          regular={true}
          style={
            Object {
              "marginBottom": 10,
              "marginLeft": 25,
              "marginRight": 25,
              "marginTop": 0,
            }
          }
        >
          <Styled(Input)
            error={false}
            label="Confirm"
            onChangeText={[Function]}
            placeholder="Enter Password Again"
            secureTextEntry={true}
            value="confirm"
          />
        </Styled(Item)>
        <Styled(Button)
          block={true}
          name="email"
          onPress={[Function]}
          padder={true}
          style={
            Object {
              "backgroundColor": "#C0B083",
              "marginBottom": 25,
              "marginLeft": 88,
              "marginRight": 100,
              "marginTop": 20,
              "width": "50%",
            }
          }
        >
          <Styled(Icon)
            name="send"
          />
          <Styled(Text)>
            Sign Up With Email
          </Styled(Text)>
        </Styled(Button)>
      </Styled(Form)>
    </Styled(Content)>
  </Image>
</Styled(Container)>
`;
